operator/webhook/app.py

```diff
@@ -9,37 +11,85 @@
 from starlette.status import HTTP_400_BAD_REQUEST
 
 from webhook import config as cfg
-from webhook.introute.sync import sync
+from webhook.core.sync import sync
+from webhook.addons.certmanager.main import sync_certificate
 from webhook.logconf import LOG_CONF
 
 _LOGGER = logging.getLogger(__name__)
 
+COMPOSITE_CONTROLLER = "CompositeController"
+
+DECORATOR_CONTROLLER = "DecoratorController"
+
+INTEGRATION_ROUTES_RESOURCE = "integrationroutes"
+
+CERTIFICATES_RESOURCE = "certificates"
+
 
 async def webhook(request: Request):
```

Comment:
I think we might be over-complicating here. As far as this method is concerned, it should really just be dealing with  serializing/deserializing the JSON, and providing a catch-all for errors that bubble up.

We could simplify by treating the request body as opaque and passing that to the appropriate sync handlers. Each handler should already know the type of request it receives (Composite vs Decorator), so we can let them pick out the specific fields, rather than do it here. Plus they might need other fields in the future, so it makes sense to pass along the full request.

Reply:
Fixed